src/Microsoft.AspNet.OData.Shared/Builder/DerivedTypeConstraintSet.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.UriParser;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    internal class DerivedTypeConstraintSet : HashSet<Type>
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public DerivedTypeConstraintSet()
+        {
+        }
+
+        public DerivedTypeConstraintSet(Type baseType)
+        {
+            ClrBaseType = baseType;
+        }
+
+        public Type ClrBaseType { get; set; }
```

Comment:
>ClrBaseType [](start = 20, length = 11)

is there any value to exposing this property? Hopefully it wouldn't ever be set/retrieved after the DerivedTypeConstraintSet was constructed (and certainly couldn't be changed once set?)

Reply:
I have to expose it mainly because of the operation configuration. We allow setting the return type after constructing the operation configuration object. 

---
In reply to: [384820274](https://github.com/OData/WebApi/pull/1977#discussion_r384820274) [](ancestors = 384820274)

Reply:
+1, If a property value constructs from Ctor, don't provide the public set for that property.
Maybe consider to change your operation configuration.

---
In reply to: [384820274](https://github.com/OData/WebApi/pull/1977#discussion_r384820274) [](ancestors = 384820274)